src/object/srv_obj_migrate.c

```diff
@@ -1326,6 +1327,19 @@ migrate_dkey(struct migrate_pool_tls *tls, struct migrate_one *mrone)
 		D_GOTO(obj_close, rc);
 	}
 
+	/* punch the object */
+	if (mrone->mo_obj_punch_eph) {
+		rc = vos_obj_punch(cont->sc_hdl, mrone->mo_oid,
+				   mrone->mo_obj_punch_eph,
+				   tls->mpt_version, VOS_OF_REPLAY_PC,
+				   NULL, 0, NULL, NULL);
```

Comment:
@jolivier23  is it ok for VOS to punch the same object with same epoch multiple times?

Reply:
It should be ok with the VOS_OF_REPLAY_PC flag.  If it doesn't work, it's a bug